src/js/vic-v2/components/PhotoField.jsx

```diff
@@ -150,15 +147,60 @@ export default class PhotoField extends React.Component {
 
   onZoom = (e) => {
     const newZoomValue = e.detail.ratio;
-    this.setState({ zoomValue: newZoomValue }, () => {
-      this.refs.slider.value = this.state.zoomValue;
-    });
+    if (newZoomValue < MAX_RATIO && newZoomValue > MIN_RATIO) {
+      let warningMessage;
+      if (this.state.zoomValue >= WARN_RATIO) {
+        warningMessage = 'Zooming in this close will make your ID picture less clear.';
+      }
+      return this.setState({ zoomValue: newZoomValue, warningMessage }, () => {
+        this.refs.slider.value = this.state.zoomValue;
+      });
+    }
+    return e.preventDefault();
+  }
+
+  setCropBox = () => {
+    const cropper = this.refs.cropper;
+    const slider = this.refs.slider;
+    setTimeout(() => {
+      const { width, naturalWidth } = this.refs.cropper.getImageData();
+      slider.value = width / naturalWidth;
+      const containerData = cropper.getContainerData();
+      const containerWidth = containerData.width;
+      const smallScreen = isSmallScreen(this.state.windowWidth);
+      const cropBoxSize = smallScreen ? 240 : 300;
+      const cropBoxLeftOffset = (containerWidth - cropBoxSize) / 2;
+      const cropBoxData = {
+        left: cropBoxLeftOffset,
+        top: 0,
+        width: cropBoxSize,
+        height: cropBoxSize
+      };
+      cropper.setCropBoxData(cropBoxData);
+    }, 0);
   }
 
+
   getErrorMessage = () => {
     return this.state.errorMessage;
   }
 
+  detectWidthAndDrag = () => {
+    const windowWidth = window.innerWidth;
+    const div = document.createElement('div');
+    const supportsDragAndDrop = ('draggable' in div) || ('ondragstart' in div && 'ondrop' in div);
+    const iOS = !!navigator.userAgent.match('iPhone OS') || !!navigator.userAgent.match('iPad');
+    const dragAndDropSupported = supportsDragAndDrop && window.FileReader;
+    const previousWidth = this.state.windowWidth;
+    this.setState({ windowWidth, dragAndDropSupported: dragAndDropSupported && !iOS }, () => {
+      const newWidth = this.state.windowWidth !== previousWidth;
```

Comment:
I wonder if the two setState callbacks would be better implemented in `componentWillReceiveProps`.

Reply:
Excellent idea, but since componentWillReceiveProps won't be fired on setState, we can use componentWillUpdate to consolidate this logic instead. Changed.